packages/ai-terminal/src/browser/ai-terminal-agent.ts

```diff
@@ -0,0 +1,189 @@
+import { ILogger } from '@theia/core';
+import { inject, injectable } from '@theia/core/shared/inversify';
+import { z } from 'zod';
+import zodToJsonSchema from 'zod-to-json-schema';
+
+const Commands = z.object({
+    commands: z.array(z.string()),
+});
+type Commands = z.infer<typeof Commands>;
+
+@injectable()
+export class AiTerminalAgent implements Agent {
+
+    id = 'Terminal Assistant';
+    name = 'Terminal Assistant';
+    description = 'This agent provides assistance to write and execute arbitrary terminal commands. \
+        Based on the user\'s request, it suggests commands and allows the user to directly paste and execute them in the terminal. \
+        It accesses the current directory, environment and the recent terminal output of the terminal session to provide context-aware assistance';
+    variables = [];
+    promptTemplates = [
+        {
+            id: 'ai-terminal:system-prompt',
+            name: 'AI Terminal System Prompt',
+            description: 'Prompt for the AI Terminal Assistant',
+            template: `
+# Instructions
+Generate one or more command suggestions based on the user's request, considering the shell being used,
+the current working directory, and the recent terminal contents. Provide the best suggestion first,
+followed by other relevant suggestions if the user asks for further options. 
+
+Parameters:
+- user-request: The user's question or request.
+- shell: The shell being used, e.g., /usr/bin/zsh.
+- cwd: The current working directory.
+- recent-terminal-contents: The last 0 to 50 recent lines visible in the terminal.
+
+Return the result in the following JSON format:
+{
+  "commands": [
+    "best_command_suggestion",
+    "next_best_command_suggestion",
+    "another_command_suggestion"
+  ]
+}
+
+## Example
+user-request: "How do I commit changes?"
+shell: "/usr/bin/zsh"
+cwd: "/home/user/project"
+recent-terminal-contents:
+git status
+On branch main
+Your branch is up to date with 'origin/main'.
+nothing to commit, working tree clean
+
+## Expected JSON output
+\`\`\`json
+\{
+  "commands": [
+    "git commit",
+    "git commit --amend",
+    "git commit -a"
+  ]
+}
+\`\`\`
+`
+        },
+        {
+            id: 'ai-terminal:user-prompt',
+            name: 'AI Terminal User Prompt',
+            description: 'Prompt that contains the user request',
+            template: `
+user-request: {{userRequest}}
+shell: {{shell}}
+cwd: {{cwd}}
+recent-terminal-contents:
+{{recentTerminalContents}}
+`
+        }
+    ];
+    languageModelRequirements: LanguageModelRequirement[] = [
+        {
+            purpose: 'suggest-terminal-commands',
+            identifier: 'openai/gpt-4o',
+        }
+    ];
+
+    @inject(LanguageModelRegistry)
+    protected languageModelRegistry: LanguageModelRegistry;
+
+    @inject(PromptService)
+    protected promptService: PromptService;
+
+    @inject(ILogger)
+    protected logger: ILogger;
+
+    async getCommands(input: {
```

Comment:
What's wrong with just having method parameters instead of a parameter object?

Reply:
Nothing wrong. It just already has a large number of parameters. When adding even more, especially should some become optional, then a long list of parameters is unwieldy

Reply:
Since there are no optional fields in the object, you get the exact same list, except it's more confusing. And it's not usually done in Theia, as far as I can tell.